pkg/skaffold/runner/context/context.go

```diff
@@ -32,7 +33,8 @@ type RunContext struct {
 	Opts *config.SkaffoldOptions
 	Cfg  *latest.Pipeline
 
-	Trigger chan bool
+	BuildTrigger  chan bool
```

Comment:
Is the plan for having a separate channel per trigger type?
In dev mode, what happens if you invoke deploy before build is complete?
would it be ignored? or queued later?

Reply:
yep, one channel per trigger.

if you send a deploy trigger during the build (or even before it), it will get queued into the channel. each channel is of size 1, so subsequent triggers sent from the client will be ignored, and the response to the user will tell them that a trigger is already queued. I debated having a way to remove a queued trigger (through a separate "cancel") endpoint, but ultimately decided this isn't really necessary, the user can just ctrl+c if they really want to stop the deploy.

Reply:
> yep, one channel per trigger.

Ack.  If we do expose multiple triggers, then maybe it would make sense to have a single chan string of x length where the string represent trigger type. 
It might make the API server code harder to check if a trigger type is enabled or not. (might need some extra server config)
e.g.:
```
In Skaffold, 
trigger := make(chan string, 3)
# In GRPC server. 
func (s *server) Build(context.Context, *empty.Empty) (*proto.ApiResponse, error) {
if s.trigger does not contain "build"; then:
   s.trigger <- "build"
else:
  return nil, fmt.Errorf("build is already queued")   
```


And makes sense for "not having a cancel trigger"